analysis/os_reports/04_eol_WP2_quality_indicators.R

```diff
@@ -256,261 +59,454 @@ df <- read_csv(file = here::here("output", "os_reports", "input_os_reports.csv.g
\ No newline at end of file
+acp_cod_quarters_rounded <- df %>%
+  group_by(study_quarter, codgrp) %>%
+  summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+  bind_rows(df %>%
+              group_by(study_quarter) %>%
+              summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+              mutate(codgrp = "All")) %>%
+  dplyr::mutate(across(.cols = all_of(cols_of_interest), .fns = ~ replace(.x, (. <= 7 & .  > 0), NA))) %>% 
+  dplyr::mutate(across(.cols = all_of(cols_of_interest), .fns = ~ .x %>% `/`(5) %>% round()*5)) %>%
+  mutate(proportion = round(count / total*100,1))
+
+fwrite(acp_cod_quarters_rounded, here::here("output", "os_reports", "WP2_quality_indicators", "acp_cod_quarters_rounded.csv"))
+
+#Raw estimates
+
+acp_cod_quarters_raw <- df %>%
+  group_by(study_quarter, codgrp) %>%
+  summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+  bind_rows(df %>%
+              group_by(study_quarter) %>%
+              summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+              mutate(codgrp = "All")) %>%
+  mutate(proportion = round(count / total*100,1))
+
+fwrite(acp_cod_quarters_raw, here::here("output", "os_reports", "WP2_quality_indicators", "acp_cod_quarters_raw.csv"))
+
+#Monthly breakdown
+
+#Rounded estimates
+
+acp_cod_months_rounded <- df %>%
+  group_by(study_month, codgrp) %>%
+  summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+  bind_rows(df %>%
+              group_by(study_month) %>%
+              summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+              mutate(codgrp = "All")) %>%
+  dplyr::mutate(across(.cols = all_of(cols_of_interest), .fns = ~ replace(.x, (. <= 7 & .  > 0), NA))) %>% 
+  dplyr::mutate(across(.cols = all_of(cols_of_interest), .fns = ~ .x %>% `/`(5) %>% round()*5)) %>%
+  mutate(proportion = round(count / total*100,1))
+
+fwrite(acp_cod_months_rounded, here::here("output", "os_reports", "WP2_quality_indicators", "acp_cod_months_rounded.csv"))
+
+#Raw estimates
+
+acp_cod_months_raw <- df %>%
+  group_by(study_month, codgrp) %>%
+  summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+  bind_rows(df %>%
+              group_by(study_month) %>%
+              summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE), total = n()) %>%
+              mutate(codgrp = "All")) %>%
+  mutate(proportion = round(count / total*100,1))
+
+fwrite(acp_cod_months_raw, here::here("output", "os_reports", "WP2_quality_indicators", "acp_cod_months_raw.csv"))
+
+#Average number of care plan codes recorded per person during the last 90 days (3 months) of life by place of death
+
+#Quarterly breakdown 
+
+#Rounded estimates
+
+acp3m_pod_quarters_rounded <- df %>%
+  group_by(study_quarter, pod_ons_new) %>%
+  summarise(count = sum(has_careplan == 'TRUE', na.rm = TRUE),
```

Comment:
does this mean the mean number of times ACP is coded for in the group of people who have ACP recorded? would we not want the mean to include people who didn't have an ACP recorded as well?

Reply:
Hmmm my logic was that we have a sense of what proportion have ACP recorded anyway, and if we included those who had no ACP in this measure, we wouldn't have a correct sense of how many records of advance care planning appear for those with at least one mention of it